site/docs/concepts/architecture/control-plane.md

```diff
@@ -8,75 +8,99 @@ sidebar_position: 3
-2. When changes are detected, it updates the Envoy Gateway configuration
-3. Envoy Gateway then updates the Envoy Proxy configuration
-4. The data plane (Envoy Proxy) processes AI traffic based on this configuration
+2. When changes are detected, it updates/generates the Envoy Gateway configuration
+3. Envoy Gateway then updates the Envoy Proxy configuration while talking to the Envoy AI Gateway controller to fine-tune the configuration.
+4. The data plane (Envoy Proxy) processes AI traffic based on this configuration where the AI Gateway ExtProc runs as a sidecar to handle AI-specific processing. The sidecar container is inserted by the AI Gateway controller into the Envoy Proxy Pod.
 
 This architecture ensures a clear separation of concerns, where the AI Gateway controller focuses on AI-specific configuration while leveraging Envoy Gateway for general proxy management.
 
 ## Components
 
-### AI Gateway Controller
+### 1. Envoy AI Gateway Controller
 The AI Gateway Controller manages AI-specific components and configurations:
 
 #### ExtProc Management
-- Deploys and configures the External Processor (ExtProc) service
-- Creates and updates ExtProc ConfigMaps with processing rules
-- Configures ExtProc security policies and authentication
-- Manages ExtProc deployments and their lifecycle
+- Creates and updates ExtProc Secrets with processing rules as well as credentials
+- Inserts the AI Gateway ExtProc as a sidecar container in the Envoy Proxy Pod via the [Kubernetes Admission Webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/). The container mounts the ExtProc config secret and communicates with the Envoy Proxy to process AI traffic.
 
 #### Resource Management
 - Watches AI Gateway Custom Resources (CRs)
-- Creates and manages `EnvoyExtensionPolicy` resources
-- Configures `HTTPRoute` resources for request routing
-- Manages backend security policies and authentication
+- Creates and manages `HTTPRoute` and `EnvoyExtensionPolicy` resources
+- Manages backend security policies and authentication, including the credentials rotation
 
 #### Integration with Envoy Gateway
 - Works alongside Envoy Gateway Controller (not directly configuring Envoy)
-- Creates resources that Envoy Gateway translates into Envoy configuration
-- Manages AI-specific extensions and filters
+- Creates resources, such as `HTTPRoute`, that Envoy Gateway translates into Envoy configuration with AI-specific processing rules
+- Serves the [Envoy Gateway Extension Server](https://gateway.envoyproxy.io/docs/tasks/extensibility/extension-server/) to fine-tune the Envoy configuration (xDS)
 - Enables token-based rate limiting through metadata
 
-### Envoy Gateway Controller
+### 2. Envoy Gateway Controller
 - Manages the core Envoy configuration through xDS
 - Handles service discovery and load balancing
 - Manages TLS certificates
 - Translates Gateway API resources into Envoy configuration
 
-## Configuration Flow
```

Comment:
Deleted this section as it seems to overlaps with the above diagram and the content seems a bit too obvious